### PR TITLE
Always allow app CRs with a deletion timestamp

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/11656

If the deletion timestamp is set on the app CR we skip the validation. This covers when app CRs are deleted directly or by deleting the namespace they belong to.

### CODE REVIEW

File: helm/app-admission-controller/templates/_helpers.tpl [hunk 1]
```diff
@@ -19,7 +19,7 @@ Common labels
 {{- define "labels.common" -}}
 app: {{ include "name" . | quote }}
 {{ include "labels.selector" . }}
-app.giantswarm.io/branch: {{ .Values.project.branch | quote }}
+app.giantswarm.io/branch: {{ .Values.project.branch | replace "#" "-" | replace "/" "-" | replace "." "-" | trunc 63 | trimSuffix "-" | quote }}
```
Comment (rossf7) [inline helm/app-admission-controller/templates/_helpers.tpl:22]: Sneaking this in. So the integration tests will work with dependabot PRs.

**Comment by rossf7** [comment]
Tested on `ginger`.

File: pkg/app/validate_app.go [hunk 2]
```diff
@@ -78,6 +78,13 @@ func (v *Validator) Validate(request *v1beta1.AdmissionRequest) (bool, error) {
 
 	v.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("validating app %#q in namespace %#q", app.Name, app.Namespace))
 
+	// We check the deletion timestamp rather than for a delete event because
+	// app CRs may be deleted by deleting the namespace they belong to.
+	if !app.DeletionTimestamp.IsZero() {
+		v.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("admitted deleted app %#q in namespace %#q", app.Name, app.Namespace))
```
Comment (tomahawk28) [inline pkg/app/validate_app.go:84]: WDYT?
```suggestion
		v.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("admitted deletion of app %#q in namespace %#q", app.Name, app.Namespace))
```

**Review by tomahawk28 — APPROVED** [review]

File: pkg/app/validate_app.go (hunk 2 above)
Comment (rossf7) [reply pkg/app/validate_app.go:84]: Much better. Thx!

**Pull request merged**